Lib/importlib/metadata.py

```diff
@@ -0,0 +1,271 @@
+import os
+import re
+import abc
+import sys
+import email
+import itertools
+import contextlib
+
+from configparser import ConfigParser
+from importlib import import_module
+from importlib.resources import Package
+from pathlib import Path
+from types import ModuleType
+from typing import Callable
+from zipfile import ZipFile
+
+
+__all__ = [
+    'Distribution',
+    'PackageNotFoundError',
+    'distribution',
+    'entry_points',
+    'resolve',
+    'version',
+    ]
+
+
+class PackageNotFoundError(ModuleNotFoundError):
+    """The package's metadata was not found."""
+
+
+class Distribution:
+    """A Python distribution package."""
+
+    @abc.abstractmethod
+    def load_metadata(self, name):
+        """Attempt to load metadata given by the name.
+
+        :param name: The name of the distribution package.
+        :return: The metadata string if found, otherwise None.
```

Comment:
This docstring style doesn't follow either PEP 8 or 257.

Reply:
Address style concerns upstream.